app/services/clicks_extractor.rb

```diff
@@ -0,0 +1,31 @@
+class ClicksExtractor < Extractor
+
+  def call
+    clicks = clicks_by_country
+
+    clicks = clicks.where('clicks.occurred_at >= ?', @params[:from]) if @params[:from].present?
+    clicks = clicks.where('clicks.occurred_at <= ?', @params[:to]) if @params[:to].present?
+
+    { total: total(clicks), clicks: paginated(clicks) }
+  end
+
+  private
+
+  def total(clicks)
+    clicks.joins(:link).where(links: { user_id: @user.id }).length
+  end
+
+  def paginated(clicks)
+    clicks.joins(:link)
+          .where(links: { user_id: @user.id })
+          .limit(size).offset(page * size)
+  end
+
+  def clicks_by_country
+    if @params[:country].present?
+      Click.by_country('%'.concat(@params[:country].capitalize.concat('%')))
```

Comment:
Here you prepare a value for sql `LIKE` statement. Instead of capitalizing param here you can just use case-insensitive version of `LIKE` -> `ILIKE`. Example: `WHERE field ILIKE '%value%'` Will match rows with 'value', 'Value', 'VALUE', 'VaLuE', etc.